ti/instrumentation/utils.go

```diff
@@ -264,6 +266,136 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger,
+		if err != nil {
+			fmt.Errorf("failed to run bazel query, error encountered %v: ", err)
+		}
+
+		count, err := strconv.Atoi(strings.TrimSpace(string(countAllJavaSrcs)))
+		if err != nil {
+			count = 0
+		}
+
+		directory := filepath.Join(workspace, "/", strings.Replace(file.Name, "/BUILD.bazel", "", -1))
+		//if no srcs present in the bazel build file, then select all files under this directory as changed
+		if count == 0 {
+			message := fmt.Sprintf("No src detected in Bazel %v, selecting all java files as changed files inside the directory %v", file.Name, directory )
+			log.Infoln(message)
+			changedRes, err = getAllJavaFilesInsideDirectory(directory, changedRes, file, uniqueFiles)
+		} else if count >= threshold {
+			//if count of files changed due to bazel build file modification is over the limit set then update the module list with test target
+			message := fmt.Sprintf("%v Files changed after changing bazel file %v; crosses limit set for bazelFileCount", count, file.Name )
+			log.Infoln(message)
+			if javaTestKindOutput != nil {
+				splitName := strings.Split(file.Name, "/")
+				moduleName := splitName[0]
+				moduleList = append(moduleList, moduleName)
+			}
+		} else {
+			//get list of .java files from the output
+			javaFileNames := extractJavaFilesFromQueryOutput(allJavaSrcsOutput)
+			for _, name := range javaFileNames {
+				//to prevent duplicate files
+				if _, exists := uniqueFiles[name]; !exists {
+					changedRes = append(changedRes, ti.File{Status: file.Status, Name: name})
+					uniqueFiles[name] = struct{}{}
+				}
+			}
+		}
+	}
+	return changedRes, moduleList, nil
+}
+
+// takes bazelQueryOutput string and parses it to extract .java files and returns in a list
+func extractJavaFilesFromQueryOutput(bazelOutput []byte) []string {
+	var javaFileNames []string
+	// Outer Regex to extract only srcs
+	pattern1 := `srcs = \[[^\]]+\]`
+	r1 := regexp.MustCompile(pattern1)
+
+	sections := r1.FindAllString(string(bazelOutput), -1)
+	for _, section := range sections {
+		// Inner Regex to match only .java files inside each src
+		pattern2 := `"//[^"]+\.java"`
+		r2 := regexp.MustCompile(pattern2)
+		matches := r2.FindAllString(section, -1)
+		for _, match := range matches {
+			srcFile := strings.Split(strings.Trim(match, `\"`), ":")
+			changedFile := strings.TrimLeft(string(srcFile[0]), "//") + "/" + string(srcFile[1])
+			javaFileNames = append(javaFileNames, changedFile)
+		}
+	}
+	return javaFileNames
+}
+
+// takes file name, runs bazel queries extract src globs defined in java rules, viz java_library and java_test
+func getJavaRulesFromBazel(file string, workspace string, ctx context.Context) ([]byte, []byte, []byte, error) {
+	bazelQueryInput := strings.Replace(file, "/BUILD.bazel", "", -1)
+	//query to fetch test src globs from java rule
+	c1 := fmt.Sprintf("cd %s; %s query 'kind(\"java\", %s:*)' --output=build | grep 'srcs =' ", workspace, bazelCmd, bazelQueryInput)
```

Comment:
Add an example of how this command looks in plain text. Make this change everywhere you're running a query.